### ISSUE TITLE
Secure cookie and Istio egress

### ISSUE THREAD

**Issue by louiscryan** [istio/istio#6548]
@msogrin commented on [Fri Feb 16 2018](https://github.com/istio/old_issues_repo/issues/211)

**Is this a BUG or FEATURE REQUEST?:**  BUG

**Did you review https://istio.io/help/ and existing issues to identify if this is already solved or being worked on?**: Yes

**What Version of Istio and Kubernetes are you using, where did you get Istio from, Installation details**
```
istioctl version
Version: 0.5.1
GitRevision: 30acfe6528107ea333543309095659b93364b30d
User: root@2e4a18076b04
Hub: docker.io/istio
GolangVersion: go1.9
BuildStatus: Clean
kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:54Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2018-01-26T19:04:38Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Is Istio Auth enabled or not ?** not

**What happened**: Using Istio egress rule to access HTTPS service that uses secure cookie authentication (such as Cloudant) fails due to the client code being forced to use HTTP protocol (according to https://istio.io/docs/tasks/traffic-management/egress.html#make-requests-to-the-external-services).

According to https://tools.ietf.org/html/rfc6265#section-4.1.2.5

> When a cookie has the Secure attribute, the user agent will include the cookie in an HTTP request only if the request is transmitted over a secure channel (typically HTTP over Transport Layer Security (TLS).

So the standards abiding clients (in this case, it's Python builtin http.cookiejar module) will discard the secure cookie when doing non-HTTPS requests, thus breaking the authentication mechanism.

**What you expected to happen:** A mechanism that allowed to use Istio egress and HTTPS connection from the client.

**How to reproduce it:**

---

@vadimeisenbergibm commented on [Tue Feb 20 2018](https://github.com/istio/old_issues_repo/issues/211#issuecomment-366973636)

@msogrin Currently, the only way to support HTTPS connection from the client is by specifying the protocol as TCP.

**Comment by vadimeisenbergibm** [comment]
@msogrin Note that Istio 0.8.0 supports HTTPS (originated by the application), no requirement to use HTTP anymore, see the second example in https://istio.io/docs/tasks/traffic-management/egress/#configuring-the-external-services .

**Comment by vadimeisenbergibm** [comment]
Resolved in 0.8.0, closing.
